File: app/src/main/java/com/andrewtse/kfgit/ui/module/main/fragment/FragmentsDataGenerator.java
package com.andrewtse.kfgit.ui.module.main.fragment;

import androidx.fragment.app.Fragment;

/**
 * @author xk
 * @date 2019/2/19
 */
public class FragmentsDataGenerator {

    public static Fragment[] getFragments() {
        Fragment[] fragments = new Fragment[4];
        fragments[0] = TrendingFragment.newInstance();
        fragments[1] = SearchFragment.newInstance();
        fragments[2] = StarredFragment.newInstance();
        fragments[3] = ProfileFragment.newInstance();
        return fragments;
    }
}

File: app/src/main/java/com/andrewtse/kfgit/ui/module/main/fragment/FragmentsDataGenerator.kt
package com.andrewtse.kfgit.ui.module.main.fragment

import androidx.fragment.app.Fragment

/**
 * @author xk
 * @date 2019/2/19
 */
open class FragmentsDataGenerator {

    companion object {
        fun getFragments(): Array<Fragment> {
            return arrayOf(TrendingFragment.newInstance(), SearchFragment.newInstance(), StarFragment.newInstance(), ProfileFragment.newInstance())
        }
    }
}